src/hmm/transition-model.h

```diff
@@ -315,7 +323,7 @@ inline int32 TransitionModel::TransitionIdToPdf(int32 trans_id) const {
   KALDI_ASSERT(static_cast<size_t>(trans_id) < id2state_.size() &&
                "Likely graph/model mismatch (graph built from wrong model?)");
   int32 trans_state = id2state_[trans_id];
-  return triples_[trans_state-1].pdf;
```

Comment:
There is a problem in this function.  It needs to check whether it's a self-loop pdf, and possibly return self_loop_pdf.  BTW, I think we should consider changing the 'pdf' in the tuples and in the HMM-states to 'forward_pdf'.  This might be clearer.  What do you think?